sqlglot/tokens.py

```diff
@@ -954,7 +955,11 @@ class Tokenizer(metaclass=_Tokenizer):
     # Handle numeric literals like in hive (3L = BIGINT)
     NUMERIC_LITERALS: t.Dict[str, str] = {}
 
-    COMMENTS = ["--", ("/*", "*/")]
+    COMMENTS = [
+        "--",
+        ("/*", "*/"),
+        ("/*+", "*/"),
+    ]
```

Comment:
Since `/*+` was previously treated as a hint for all dialects, we don't have to add it in `COMMENTS`, but rather do it in the metaclass. That way you don't have to change MySQL, for example. It would look like this:

```python
        klass._COMMENTS = {
            **dict(
                (comment, None) if isinstance(comment, str) else (comment[0], comment[1])
                for comment in klass.COMMENTS
            ),
            "{#": "#}",  # Ensure Jinja comments are tokenized correctly in all dialects
            "/*+": "*/",  # Ensure hints are treated as comments and tokenized correctly in all dialects
        }
```